src/support/Type.ts

```diff
@@ -4,13 +4,25 @@ import { StringHelper } from './StringHelper';
 import { TypeMetadata } from './metadata/TypeMetadata';
 import { Entity } from '../Entity';
 
-export function Type(type?: Function, jsonKey?: string) {
+// Types that can be passed as first argument to `EntityBuilder`.
+export type Buildable = typeof Entity | ({ new(): any });
```

Comment:
```suggestion
export type Buildable = typeof Entity | ({ new<T extends Entity>(): T });
```

Can you build a non-entity – and if so, can we type it?

Reply:
Well, technically anything new-able can be built through EntityBuilder, so I'm not sure if it's a good idea to try to come up with a comprehensive list.